astropy/io/fits/tests/test_header.py

```diff
@@ -1384,8 +1384,32 @@ def test_header_insert_before_keyword(self):
         assert list(header.keys())[-3] == 'TEST1'
 
     def test_remove(self):
-        # TODO: Test the Header.remove() method; add support for ignore_missing
-        pass
+        header = fits.Header([('A', 'B'), ('C', 'D')])
+
+        #When keyword is present in the header it should be removed.
+        header.remove('C')
+        assert len(header) == 1
+        assert list(header) == ['A']
+        assert 'C' not in header
+
+        #When keyword is not present in the header and ignore_missing is False, KeyError should be raised
+        try:
+            header.remove('F')
+        except KeyError as e:
+            assert e.args[0] == "Keyword 'F' not in the header."
```

Comment:
Instead do
```python
with pytest.raises(KeyError):
    header.remove('F')
```
to test that the exception is actually raised.  Your test, as written, will run with the exception *is* raised, but will silently fail when the exception is *not* raised, which is a bug.

If you want to test the text of the exception message you can do:

```python
with pytest.raises(KeyError) as exc:
    header.remove('F')
    # The exception is stored in exc and you can test against it
```
but I don't think it's that important in this case.  Either way.